src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Acquisition/MissingSetupComponentRegistrationService.cs

```diff
@@ -207,7 +194,13 @@ void AddConfiguration()
             // Fall back to the full path of the project if the project GUID has not yet been set.
             if (projectGuid == Guid.Empty)
             {
-                projectConfigurationSet = ProjectPathToProjectConfigurationsMap.GetOrAdd(project.UnconfiguredProject.FullPath, guid => new ConcurrentHashSet<ProjectConfiguration>());
+                // This collection is not commonly needed, so we construct it lazily.
+                if (_projectPathToProjectConfigurationsMap is null)
+                {
+                    Interlocked.CompareExchange(ref _projectPathToProjectConfigurationsMap, new(StringComparers.Paths), null);
```

Comment:
Maybe explicitly name the arguments? It might just be me but I always get tripped up by the order of the second and third arguments.

Reply:
I also get tripped up on this.

However in this case this code went away. We still have 9 other usages of `Interlocked.CompareExchange` in this repo, none of which have named arguments.